jupyter_server_fileid/manager.py

```diff
@@ -120,30 +124,35 @@ def __init__(self, *args, **kwargs):
         self.con.execute("PRAGMA journal_mode = WAL")
         self.con.execute(
             "CREATE TABLE IF NOT EXISTS Files("
-            "id INTEGER PRIMARY KEY AUTOINCREMENT, "
+            "id TEXT PRIMARY KEY NOT NULL, "
             "path TEXT NOT NULL UNIQUE"
             ")"
         )
         self.con.execute("CREATE INDEX IF NOT EXISTS ix_Files_path ON Files (path)")
         self.con.commit()
 
-    def index(self, path: str) -> int:
+    def _create(self, path: str) -> str:
+        id = self._uuid()
+        self.con.execute("INSERT INTO Files (id, path) VALUES (?, ?)", (id, path))
+        return id
+
+    def index(self, path: str) -> str:
         row = self.con.execute("SELECT id FROM Files WHERE path = ?", (path,)).fetchone()
         existing_id = row and row[0]
 
         if existing_id:
             return existing_id
 
         # create new record
-        cursor = self.con.execute("INSERT INTO Files (path) VALUES (?)", (path,))
+        id = self._create(path)
         self.con.commit()
-        return cursor.lastrowid  # type:ignore
+        return id
 
-    def get_id(self, path: str) -> Optional[int]:
+    def get_id(self, path: str) -> Optional[str]:
         row = self.con.execute("SELECT id FROM Files WHERE path = ?", (path,)).fetchone()
```

Comment:
I think `ArbitraryFileIdManager` needs to use absolute paths (if `path` is not absolute already) by prepending `root_dir` or maintain another table of `root_dirs` that can then be iterated against to locate `path`.  Arbitrary instances don't have the luxury of using the filesystem as a "check".   My concern is what's going to happen when an operator restarts the server with different `root_dir` values when one is a sub-directory of another.

This should probably be fleshed out but we should consider what kinds of things someone implementing a FileIDManager would need if they wanted to be filesystem (or better yet, _storage_) agnostic.

Reply:
`ArbitraryFileIdManager` can't make any assumptions about the filesystem. It doesn't know about absolute or relative paths, as that would assume a hierarchical filesystem.

The consensus is that it's OK for the `ArbitraryFileIdManager` to not be fully fleshed out and handle cases like moving server roots or moving directories. Anything more requires assumptions about your filesystem, and that requires either `LocalFileIdManager` or some other custom implementation.